app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-integer/dynamic-content-form-field-integer.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+angular.module('bulbs.cms.dynamicContent.form.field.integer', [
+  'bulbs.cms.site.config',
+  'bulbs.cms.dynamicContent.form.input.label',
+  'bulbs.cms.dynamicContent.form.input.errors'
+])
+  .directive('dynamicContentFormFieldInteger', [
+    'CmsConfig',
+    function (CmsConfig) {
+      return {
+        require: ['ngModel', '^^form'],
+        restrict: 'E',
+        scope: {
+          name: '@',
+          ngModel: '=',
+          schema: '='
+        },
+        templateUrl: CmsConfig.buildComponentPath(
+          'dynamic-content',
+          'dynamic-content-form',
+          'dynamic-content-form-field',
+          'dynamic-content-form-field-integer',
+          'dynamic-content-form-field-integer.html'
+        ),
+        link: function (scope, element, attr, ctrls) {
+          var formField = ctrls[1][scope.name];
+          formField.$validators.integer = function (modelValue) {
```

Comment:
Given that the input is already type `number` I believe this validator comes for free out of the box: https://code.angularjs.org/1.3.19/docs/api/ng/input/input%5Bnumber%5D

Available in `$error` as `number`.